Objects/abstract.c

```diff
@@ -1714,6 +1715,53 @@ PyNumber_ToBase(PyObject *n, int base)
     return res;
 }
 
+#define NB_NAMES(op)                           \
+    [NB_##op] = NB_##op##_NAME,                \
+    [NB_INPLACE_##op] = NB_INPLACE_##op##_NAME
+
+static const char nb_names[][4] = {
+    NB_NAMES(AND),
+    NB_NAMES(FLOOR_DIVIDE),
+    NB_NAMES(LSHIFT),
+    NB_NAMES(MATRIX_MULTIPLY),
+    NB_NAMES(OR),
+    NB_NAMES(RSHIFT),
+    NB_NAMES(SUBTRACT),
+    NB_NAMES(TRUE_DIVIDE),
+    NB_NAMES(XOR),
+};
+
+#undef NB_NAMES
+
+#define NB_BINSLOT(op) [NB_INPLACE_##op] = NB_##op
+    
+static const uint8_t nb_binary_slots[] = {
+    NB_BINSLOT(AND),
+    NB_BINSLOT(FLOOR_DIVIDE),
+    NB_BINSLOT(LSHIFT),
+    NB_BINSLOT(MATRIX_MULTIPLY),
+    NB_BINSLOT(OR),
+    NB_BINSLOT(RSHIFT),
+    NB_BINSLOT(SUBTRACT),
+    NB_BINSLOT(TRUE_DIVIDE),
+    NB_BINSLOT(XOR),
+};
+
+#undef NB_BINSLOT
+
+PyObject *
+_PyNumber_Op(PyObject *o1, PyObject *o2, unsigned op)
```

Comment:
`_PyNumber_Op` and `_PyNumber_InPlaceOp` both consist of one line of code, and are used one each. 
Just delete them and use `binary_op`/`binary_iop` directly?